### PR TITLE
 [opt](Nereids) always fallback when parse failed

### PR DESCRIPTION
**Pull request by dutyu**
## Proposed changes

always fallback to legacy planner when parse failed even if enable_fallback_to_original_planner is set to false

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.6 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         18 seconds loaded 2358488459 Bytes, about 124 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17164269441 Bytes

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 57.28 seconds
 stream load tsv:          503 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17168835474 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 56.16 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.8 seconds inserted 10000000 Rows, about 335K ops/s
 storage size: 17162124940 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.32 seconds
 stream load tsv:          502 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162420918 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -421,23 +421,26 @@ private void handleQuery() {
...
-                break;
-            }
-
             parsedStmt.setOrigStmt(new OriginStatement(originStmt, i));
             parsedStmt.setUserInfo(ctx.getCurrentUserIdentity());
             executor = new StmtExecutor(ctx, parsedStmt);
             ctx.setExecutor(executor);
 
             try {
+                // support select hint e.g. select /*+ SET_VAR(query_timeout=1) */ sleep(3);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:430]: u should update class `org.apache.doris.nereids.processor.pre.EliminateLogicalSelectHint`

**Comment by morrySnow** [comment]
we already support to set `enable_fallback_to_original_planner` in hint. So, what the root problem of this PR want to fix?

**Comment by dutyu** [comment]
> we already support to set `enable_fallback_to_original_planner` in hint. So, what the root problem of this PR want to fix?

The conditional check of `enable_fallback_to_original_planner`(method `handleQuery()` at `fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java`)  is before the invoke of `analyzeVariablesInStmt`, so maybe the hint of `enable_fallback_to_original_planner` is not works fine now ?

**Comment by morrySnow** [comment]
> > we already support to set `enable_fallback_to_original_planner` in hint. So, what the root problem of this PR want to fix?
> 
> The conditional check of `enable_fallback_to_original_planner`(`handleQuery()` method at `fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java`) is before the invoke of `analyzeVariablesInStmt`, so maybe the hint of `enable_fallback_to_original_planner` does not works fine now ?

not right. enable_fallback_to_original_planner check is in StmtExecutor#executeByNereids, so it could be work fine

**Comment by dutyu** [comment]
run buildall

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.24 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17165497807 Bytes

**Review by morningman — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**